src/providers/expense/expense.js

```diff
@@ -0,0 +1,50 @@
+import AWS from 'aws-sdk/global';
+import S3 from 'aws-sdk/clients/s3';
+import Keys from '@constants/Keys';
+
+export const types = {
+  UPLOAD_RECEIPT: 'EXPENSE/UPLOAD_RECEIPT',
+};
+
+const initialState = {};
+
+export default function reducer(state = initialState, action) {
+  switch (action.type) {
+    case types.UPLOAD_RECEIPT: {
+      return {
+        ...state,
+      };
+    }
+    default:
+      return state;
+  }
+}
+
+export const expenseActions = {
+  uploadReceipts: file => (dispatch, getState) => {
+    const { selectedTrip } = getState().trip;
```

Comment:
why the getState() here? does state not get passed in here? Also, is this like a thunk action creator?

Reply:
`getState()` returns the current state